scripts/ansible/roles/windows/jenkins/tasks/slave-setup.yml

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+---
+- name: Jenkins | Set default host variables (if not already set by the user)
+  block:
+    - set_fact:
+        jenkins_agent_name: "{{ ansible_hostname }}"
+      when: jenkins_agent_name is not defined
+
+    - set_fact:
+        jenkins_agent_executors_count: 1
+      when: jenkins_agent_executors_count is not defined
+
+    - set_fact:
+        jenkins_agent_label: ""
+      when: jenkins_agent_label is not defined
+
+    - set_fact:
+        jenkins_agent_root_dir: "C:/Jenkins"
+      when: jenkins_agent_root_dir is not defined
+
+- name: Provision new slave node
+  delegate_to: localhost
+  block:
+    - name: Provision new slave node - Groovy provisioning
+      jenkins_script:
+        script: |
+            import jenkins.model.*
+            import hudson.model.*
+            import hudson.slaves.*
+            Jenkins.instance.addNode(
+              new DumbSlave(
+              "{{ jenkins_agent_name }}",
+              "",
+              "{{ jenkins_agent_root_dir }}",
+              "{{ jenkins_agent_executors_count }}",
+              Node.Mode.NORMAL,
+              "{{ jenkins_agent_label }}",
+              new JNLPLauncher(),
+              new RetentionStrategy.Always(),
+              new LinkedList()))
+
+        user: "{{ jenkins_admin_name }}"
+        password: "{{ jenkins_admin_password }}"
+        url: "{{ jenkins_url }}"
+
+    - name: Provision new slave node - Groovy slave secret extraction
+      jenkins_script:
+        script: jenkins.model.Jenkins.getInstance().getComputer("{{ jenkins_agent_name }}").getJnlpMac()
+        user: "{{ jenkins_admin_name }}"
+        password: "{{ jenkins_admin_password }}"
+        url: "{{ jenkins_url }}"
+      register: script_output
+
+    - set_fact: node_secret="{{script_output.output.lstrip('Result:') | trim}}"
+
+- name: Download slave-agent.jnlp from Jenkins Master
+  win_get_url:
+    url: "{{ jenkins_url }}/computer/{{ jenkins_agent_name }}/slave-agent.jnlp"
+    force_basic_auth: yes
+    url_username: '{{ jenkins_admin_name }}'
+    url_password: '{{ jenkins_admin_password }}'
+    dest: "{{ jenkins_agent_root_dir }}/slave-agent.jnlp"
+
+- name: Make sure that Jenkins home exists
+  win_file:
+    path: "{{ jenkins_agent_root_dir }}"
+    state: directory
+
+- name: Download agent.jar from Jenkins Master
+  win_get_url:
+    url: "{{ jenkins_url }}/jnlpJars/agent.jar"
+    force_basic_auth: yes
+    url_username: "{{ jenkins_admin_name }}"
+    url_password: "{{ jenkins_admin_password }}"
+    dest: "{{ jenkins_agent_root_dir }}/agent.jar"
+
+- name: Download all the installers
+  win_get_url:
+    url: "{{ item.value.url }}"
+    dest: "{{ item.value.dest }}"
+    timeout: 60
+  retries: 5
+  with_dict: "{{ packages }}"
+
+- name: Add all the packages to the machine PATH
+  win_path:
+    elements: "{{ item }}"
```

Comment:
I think that it works if we directly pass `"{{ packages_env_path }}"` here instead of using a loop with `with_items`.

Reply:
Fixed